### PR TITLE
Lower the root logger's log level to --log-level instead of to NOTSET

### LINKED ISSUE #2977: Pytest 3.3: root logger has wrong logging level
In Pytest 3.3, the following test case fails:

```class TestLogging(unittest.TestCase):
    def test_default_level(self):
        self.assertEqual(logging.getLogger().getEffectiveLevel(),
                         logging.WARNING)
```
It passes in Pytest 3.2. The Python documentation (https://docs.python.org/3.6/library/logging.html#logging.Logger.setLevel) says that the root logger is created with level WARNING, so I believe this is a bug in pytest.

I think this is likely related to #2974 .

### PR DESCRIPTION
**Pull request by segevfiner**
`logging` is a seriously confusing module, I'm not really sure that's the right way to fix this, so review with care.

Fixes #2977

## Checklist
Thanks for submitting a PR, your contribution is really appreciated!

Here's a quick checklist that should be present in PRs:

- [x] Add a new news fragment into the changelog folder
  * name it `$issue_id.$type` for example (588.bugfix)
  * if you don't have an issue_id change it to the pr id after creating the pr
  * ensure type is one of `removal`, `feature`, `bugfix`, `vendor`, `doc` or `trivial`
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."
- [x] Target: for `bugfix`, `vendor`, `doc` or `trivial` fixes, target `master`; for removals or features target `features`;
- [ ] Make sure to include reasonable tests for your change if necessary

Unless your change is a trivial or a documentation fix (e.g.,  a typo or reword of a small section) please:

- [x] Add yourself to `AUTHORS`, in alphabetical order;

### CODE REVIEW

**Comment by nicoddemus** [comment]
> logging is a seriously confusing module

I agree. 😓 

It seems the tests are now failing... @segevfiner if you have the time can you perhaps take a look?

File: _pytest/logging.py [hunk 1]
```diff
@@ -242,6 +242,8 @@ def __init__(self, config):
         The formatter can be safely shared across all handlers so
         create a single one for the entire test session here.
         """
+        self.log_level = get_actual_log_level(
+            config, 'log_level') or logging.WARNING
```
Comment (nicoddemus) [inline _pytest/logging.py]: We might consider to use `None` as default; then if `None`, we don't touch the [log level](https://github.com/segevfiner/pytest/blob/b9ddb253a55bc517f83a3e4a5bed5af70bb93ddf/_pytest/logging.py#L98) at all all.

**Comment by segevfiner** [comment]
The failing tests (There might be other failures too) show why the root logger was probably set to `logging.NOTSET` in the first place. The way, *I think*, logging works when you write a message to a logger:
1. The logger finds it's log level. If the logger's level is `NOTSET` it goes upwards in the logger hierarchy looking for the closest logger that has a level that is different than `NOTSET`.
2. If the message's level is lower than the log level it doesn't get passed to *any handler at all*. This means that by setting the root logger and `LogCaptureHandler` to level `WARNING`, you get no logs lower than that level captured.
3. If the message is not dropped by the level, each handler is called, where it's own set log level get's another chance to filter the message. Handlers will be called up the hierarchy if the logger's `propogate` field is `True`.
(And this is ignoring some other nasty mechanisms such as log disabling and log filters...)

I guess that means that simply not setting the root logger's log level won't be enough since I think the module is currently expected to catch *all* logs by default.

That's what I get for underestimating `logging`... I guess this PR can be closed... 😩

File: _pytest/logging.py (hunk 1 above)
Comment (thisch) [reply _pytest/logging.py]: Note that if you pass `None` to `logger.setLevel` or `handler.setLevel` a TypeError is raised.

**Comment by nicoddemus** [comment]
Thanks anyway @segevfiner, this certaily helped to shed some more light in the subtle details involved. We continue the discussion about logging in general in #3013.

File: _pytest/logging.py (hunk 1 above)
Comment (nicoddemus) [reply _pytest/logging.py]: Indeed, but actually I meant to check for `None` before calling `setLevel`.

**Pull request closed without merging**